nums/core/array/application.py

```diff
@@ -71,6 +71,8 @@ def compute_block_shape(self,
             dtype = np.dtype(dtype)
         elif dtype in (bool, np.bool_):
             dtype = np.dtype(np.bool_)
+        elif dtype in (object, np.object_):
```

Comment:
In one of the doctests (I can't recall which one, I noticed it randomly), this would error, saying `np.object` was not a valid dtype. NumPy allows it to be used, so I added this line to fix the issue.

Reply:
I'm unable to track down which doctest it was, so for now I'll remove this.